src/MetaMorpho.sol

```diff
@@ -530,7 +554,8 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     /// @inheritdoc IERC4626
     function totalAssets() public view override(IERC4626, ERC4626) returns (uint256 assets) {
         for (uint256 i; i < withdrawQueue.length; ++i) {
-            assets += _supplyBalance(_marketParams(withdrawQueue[i]));
+            uint256 supplyAssets = _simulateSupplyBalance(_marketParams(withdrawQueue[i]));
```

Comment:
If I understand your point @Rubilmax you suggest to revert on `totalAssets` and too bad for the ERC4626, right?

Reply:
Correct! I only suggest to change `sortWithdrawQueue`